### DISCUSSION TITLE
is there support for lists overlap?

### DISCUSSION THREAD

**Discussion by ptitjano** [duckdb/duckdb#7457] [Q&A]
I'm looking to check if two lists have elements in common. The corresponding postgresql syntax is:

```sql
ARRAY[1,4,3]  &&  ARRAY[2,1]
```

Is there something similar in duckdb?

**Comment by mtreguer** [comment]
For information, the postgres documentation :  
https://www.postgresql.org/docs/11/functions-array.html

**Comment by maiadegraaf** [comment]
I'm actually working on a rewrite for a `list_intersect`, `list_has_any`, and `list_has_all` right now which should be done soon (#1940).

For now either of these more or less provide the same functionality as `&&`:
- `select list_has(list_transform([1,4,3], x -> list_contains([2,1], x)), true);`
- `select CASE WHEN len(list_filter([1,4,3], x -> list_contains([2,1], x))) > 0 THEN true ELSE false END;`

**Reply by Tishj** [reply]
We can likely also add `&&` as an alias for this

**Reply by maiadegraaf** [reply]
The PR (#7518) implementing `&&` (among other list function rewrites) has just been merged :)
